Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -0,0 +1,501 @@
+    /// Exists if `layout(options: MessageLayoutOptions)` was invoked with the options containing `.threadInfo`.
+    public private(set) var threadReplyCountButton: UIButton?
+
+    /// Shows the avatar of the user who left the latest thread reply.
+    /// Exists if `layout(options: MessageLayoutOptions)` was invoked with the options containing `.threadInfo`.
+    public private(set) var threadAvatarView: ChatAvatarView?
+
+    /// Shows the arrow from message bubble to `threadAvatarView` view.
+    /// Exists if `layout(options: MessageLayoutOptions)` was invoked with the options containing `.threadInfo`.
+    public private(set) var threadArrowView: _ThreadArrowView<ExtraData>?
+
+    // MARK: - Containers
+
+    /// Specifies the max possible width of `mainContainer`.
+    /// Should be in [0...1] range, where 1 makes the container fill the entire superview's width.
+    open var maxContentWidthMultiplier: CGFloat { 0.75 }
+
+    /// The root container which holds `authorAvatarView` (or the avatar padding) and `bubbleThreadMetaContainer`.
+    public lazy var mainContainer = ContainerStackView(axis: .horizontal)
+        .withoutAutoresizingMaskConstraints
+
+    /// The container which holds `bubbleView` (or `bubbleContentContainer` directly), `threadInfoContainer`, and `metadataView`
+    public private(set) lazy var bubbleThreadMetaContainer = ContainerStackView(axis: .vertical, spacing: 4)
+        .withoutAutoresizingMaskConstraints
+
+    /// The container which holds `quotedMessageView` and `textView`
+    public private(set) lazy var bubbleContentContainer = ContainerStackView(axis: .vertical)
+        .withoutAutoresizingMaskConstraints
+
+    /// The container which holds `threadArrowView`, `threadAvatarView`, and `threadReplyCountButton`
+    public private(set) var threadInfoContainer: ContainerStackView?
+
+    /// Makes sure the `layout(options: MessageLayoutOptions)` is called just once.
+    /// - Parameter options: The options describing the layout of the content view.
+    open func setUpLayoutIfNeeded(options: MessageLayoutOptions) {
+        guard layoutOptions == nil else {
+            assert(layoutOptions == options, "Attempt to apply different layout")
+            return
+        }
+        layout(options: options)
+        layoutOptions = options
+    }
+
+    /// Instantiates the subviews and laid them out based on the received options.
+    /// - Parameter options: The options describing the layout of the content view.
+    open func layout(options: MessageLayoutOptions) {
+        var constraintsToActivate: [NSLayoutConstraint] = []
+
+        // Main container
+        mainContainer.alignment = .axisTrailing
+        mainContainer.isLayoutMarginsRelativeArrangement = true
+        mainContainer.layoutMargins.top = 0
+
+        addSubview(mainContainer)
+        constraintsToActivate += [
+            mainContainer.topAnchor.pin(equalTo: topAnchor).with(priority: .streamAlmostRequire),
```

Comment:
We have `reactionsBubbleView` which should be pinned to the `topAnchor` when we have reactions. The mainContrinaer's top anchor should be aligned with `centerY` in this case. It's great you noticed this, maybe we can set just one constraint instead of 2 👍

Reply:
✅ Done!